pkg/targets/adapter/mongodbtarget/adapter.go

```diff
@@ -0,0 +1,250 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package mongodbtarget implements an adapter that connects to a MongoDB database
+// and allows a user to insert, query, and update documents via cloudevents.
+package mongodbtarget
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
+	"knative.dev/pkg/logging"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/zap"
+
+	"github.com/triggermesh/triggermesh/pkg/apis/targets"
+	"github.com/triggermesh/triggermesh/pkg/apis/targets/v1alpha1"
+	"github.com/triggermesh/triggermesh/pkg/metrics"
+	targetce "github.com/triggermesh/triggermesh/pkg/targets/adapter/cloudevents"
+)
+
+// NewTarget returns the adapter implementation.
+func NewTarget(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClient cloudevents.Client) pkgadapter.Adapter {
+	env := envAcc.(*envAccessor)
+	logger := logging.FromContext(ctx)
+	metrics.MustRegisterEventProcessingStatsView()
+
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(env.ServerURL))
+	if err != nil {
+		logger.Fatal("error connecting to mongodb", err)
+		return nil
+	}
+
+	replier, err := targetce.New(env.Component, logger.Named("replier"),
+		targetce.ReplierWithStatefulHeaders(env.BridgeIdentifier),
+		targetce.ReplierWithStaticResponseType(v1alpha1.EventTypeMongoDBStaticResponse),
+		targetce.ReplierWithPayloadPolicy(targetce.PayloadPolicy(env.CloudEventPayloadPolicy)))
+	if err != nil {
+		logger.Panicf("Error creating CloudEvents replier: %v", err)
+	}
+
+	mt := &pkgadapter.MetricTag{
+		ResourceGroup: targets.MongoDBTargetResource.String(),
+		Namespace:     envAcc.GetNamespace(),
+		Name:          envAcc.GetName(),
+	}
+
+	return &adapter{
+		mclient:           client,
+		defaultDatabase:   env.DefaultDatabase,
+		defaultCollection: env.DefaultCollection,
+
+		replier:  replier,
+		ceClient: ceClient,
+		logger:   logger,
+		sr:       metrics.MustNewEventProcessingStatsReporter(mt),
+	}
+}
+
+var _ pkgadapter.Adapter = (*adapter)(nil)
+
+type adapter struct {
+	mclient           *mongo.Client
+	defaultDatabase   string
+	defaultCollection string
+
+	replier  *targetce.Replier
+	ceClient cloudevents.Client
+	logger   *zap.SugaredLogger
+	sr       *metrics.EventProcessingStatsReporter
+}
+
+func (a *adapter) Start(ctx context.Context) error {
+	a.logger.Info("Starting MongoDB adapter")
+	return a.ceClient.StartReceiver(ctx, a.dispatch)
+}
+
+func (a *adapter) dispatch(event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
+	ctx := context.Background()
```

Comment:
Not sure about why the context is created at 99.
I think you could use the signature that allows passing the request's context instead (unless there is a reason I am not getting)

```suggestion
func (a *adapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
```

Reply:
I did not know you could do this and that the CE Receiver would automatically pass it. thank you.